presto-main/src/main/java/com/facebook/presto/util/Failures.java

```diff
@@ -171,6 +173,10 @@ private static ErrorCode toErrorCode(Throwable throwable)
             return SLICE_TOO_LARGE.toErrorCode();
         }
 
+        if (throwable instanceof InvalidTypeDefinitionException) {
```

Comment:
Rather than catch and redefine, let's just directly throw a `PrestoException` with `INVALID_TYPE_DEFINITION`.

Reply:
you can't throw a PrestoException from presto-common, as it's defined in presto-spi